aws-java-sdk-s3/src/main/java/com/amazonaws/services/s3/transfer/DownloadCallable.java

```diff
@@ -174,6 +174,8 @@ private void downloadInParallel(int partCount) throws Exception {
             lastFullyMergedPartNumber = 0;
         }
 
+        ServiceUtils.createParentDirectoryIfNecessary(dstfile);
```

Comment:
My initial thought was try to do this as late as possible - moving it up to line 128 doesn't actually buy us anything in this context because it'll get created down both paths (single/multi) anyway.

However - I can't think of a good reason *not* to move it up to earlier in the piece and it might even be useful from a permissions perspective - if the app doesn't have permissions to write might as well bomb before we've even started trying to download.